### PR TITLE
Graph waypoint reached

### LINKED ISSUE #173: Visualize the boundary lines for waypoint reached
Waypoint reached uses a boundary line to define when it returns true or false. It would be great to have the option to display this line in the local path visualizer to easily understand the boat's behavior.

### PR DESCRIPTION
**Pull request by patrick-5546**
Fixes #173 
Screenshot from 2021-01-23 12-32-22

### CODE REVIEW

File: test/test_Path.py [hunk 1]
```diff
@@ -41,48 +42,51 @@ def createState(spaceInformation, xy):
...
+        # Setup path from (100,1) to (1,1)
+        # Test passes when distance to next waypoint is less than or equal to WAYPOINT_REACHED_DISTANCE
+        path = self.createPathWithGivenPositionsXY([[100, 1], [1, 1]])
+        boundaryCoord = 1 + WAYPOINT_REACHED_DISTANCE
 
         # Test reachedPos
-        reachedPos = utils.XYToLatlon(xy=(0.9, -100), referenceLatlon=path.getReferenceLatlon())
+        reachedPos = utils.XYToLatlon(xy=(boundaryCoord - 0.01, -100), referenceLatlon=path.getReferenceLatlon())
         self.assertTrue(path.nextWaypointReached(reachedPos))
         self.assertTrue(path.lastWaypointReached(reachedPos))
```
Comment (patrick-5546) [inline test/test_Path.py:86]: I am having some trouble with this test. It passes when `boundaryCoord == 1`, but it should take into account `WAYPOINT_REACHED_DISTANCE` as well. Is there a way where I can see the values of the variables in path? I tried printing when running `catkin_make run_tests`, but that didn't work.

**Comment by patrick-5546** [comment]
Used `rostest local_pathfinding test_Path.test --text` so that `print()` functioned properly when running ros tests.

Also, used `abs(x-y) < 0.1` for float comparision. Is this okay?

**Comment by JoshuaJSam** [comment]
Looks good!

**Review by JoshuaJSam — APPROVED** [review]
Looks good! :)

**Pull request merged**